src/query/ast/src/ast/format/syntax/mod.rs

```diff
@@ -0,0 +1,50 @@
+// Copyright 2022 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+mod common;
+mod ddl;
+mod dml;
+mod expr;
+mod query;
+
+use common_exception::Result;
+use ddl::*;
+use dml::*;
+use pretty::RcDoc;
+use query::*;
+
+use crate::ast::Statement;
+
+pub fn format_statement_syntax(stmt: Statement, max_width: usize) -> Result<String> {
+    let mut bs = Vec::new();
+    statement_doc(stmt).render(max_width, &mut bs)?;
+    Ok(String::from_utf8(bs)?)
+}
+
+fn statement_doc(stmt: Statement) -> RcDoc {
+    match stmt {
+        Statement::Query(query) => query_doc(*query),
+        Statement::Insert(insert_stmt) => insert_doc(insert_stmt),
+        Statement::Delete {
+            table_reference,
+            selection,
+        } => delete_doc(table_reference, selection),
+        Statement::Copy(copy_stmt) => copy_doc(copy_stmt),
```

Comment:
Why deal with the `COPY/DDL` special cases here? Not using `RcDoc::text(stmt.to_string())` for them all.

Reply:
The SQL of `copy` is usually longer, using `RcDoc` can print out better-looking results.

for example
```sql
COPY INTO
    t1
FROM
    's3://mybucket/data.csv'
FILE_FORMAT = (
    field_delimiter = ",",
    record_delimiter = "\n",
    skip_header = "1",
    type = "CSV"
)
SIZE_LIMIT = 10
```

Reply:
Ok.
It would be great to explain why we handle these cases:
```
   Statement::CreateTable(create_table_stmt) => create_table_doc(create_table_stmt),
        Statement::AlterTable(alter_table_stmt) => alter_table_doc(alter_table_stmt),
        Statement::CreateView(create_view_stmt) => create_view_doc(create_view_stmt),
        Statement::AlterView(alter_view_stmt) => alter_view_doc(alter_view_stmt),
```

Reply:
done.